File: scratch/extract_strings.js
const fs = require('fs');
const path = require('path');

const FILES_TO_SCAN = [
    "c:\\Users\\LucianCebuc\\AERO\\app\\(auth)\\login.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(auth)\\register.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(auth)\\language.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\_layout.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\ride.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\request.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\searching.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\active.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\rating.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\profile.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\rides.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(passenger)\\passes.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(driver)\\_layout.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(driver)\\drive.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(driver)\\profile.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(driver)\\subscription.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(driver)\\rides.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(driver)\\passes.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(admin)\\_layout.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(admin)\\alerts.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(admin)\\dashboard.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(admin)\\drivers.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\(admin)\\settings.tsx",
    "c:\\Users\\LucianCebuc\\AERO\\app\\chat\\[rideId].tsx",
    "c:\\Users\\LucianCebuc\\AERO\\components\\feature\\DriverRequestCard.tsx",
];

const romChars = /[ășțâîĂȘȚÂÎ]/;

const extracted_strings = {};

for (const filepath of FILES_TO_SCAN) {
    if (!fs.existsSync(filepath)) {
        console.log(`Skipping missing file: ${filepath}`);
        continue;
    }
    
    console.log(`Scanning: ${filepath}`);
    const content = fs.readFileSync(filepath, 'utf8');
    
    const file_strings = new Set();
    
    // Pattern to match JSX text: >Text Content<
    // We match any > text < where text contains letters and doesn't start with { or <
    const jsxRegex = />\s*([^<{]+?[a-zA-ZăâîșțĂÂÎȘȚ].*?)\s*</g;
    let match;
    while ((match = jsxRegex.exec(content)) !== null) {
        const text = match[1].trim();
        if (text && !text.startsWith('{') && !text.endsWith('}')) {
            file_strings.add(text);
        }
    }
    
    // Pattern to match single or double quoted strings
    const strRegex = /['"]([^'"]*?[ăâîșțĂÂÎȘȚ][^'"]*?)['"]/g;
    while ((match = strRegex.exec(content)) !== null) {
        const text = match[1].trim();
        if (text) {
            file_strings.add(text);
        }
    }
    
    // Also capture template literals containing Romanian characters
    const tempRegex = /`([^`]*?[ăâîșțĂÂÎȘȚ][^`]*?)`/g;
    while ((match = tempRegex.exec(content)) !== null) {
        const text = match[1].trim();
        if (text) {
            file_strings.add(text);
        }
    }
    
    if (file_strings.size > 0) {
        extracted_strings[path.basename(filepath)] = Array.from(file_strings).sort();
    }
}

const outputPath = "c:\\Users\\LucianCebuc\\AERO\\scratch\\extracted_strings.json";
fs.mkdirSync(path.dirname(outputPath), { recursive: true });
fs.writeFileSync(outputPath, JSON.stringify(extracted_strings, null, 2), 'utf8');
console.log(`Extraction complete! Saved to ${outputPath}`);

File: scratch/extract_strings.py
import os
import re
import json

FILES_TO_SCAN = [
    r"c:\Users\LucianCebuc\AERO\app\(auth)\login.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(auth)\register.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(auth)\language.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\_layout.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\ride.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\request.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\searching.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\active.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\rating.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\profile.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\rides.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(passenger)\passes.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(driver)\_layout.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(driver)\drive.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(driver)\profile.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(driver)\subscription.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(driver)\rides.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(driver)\passes.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(admin)\_layout.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(admin)\dashboard.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(admin)\drivers.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(admin)\alerts.tsx",
    r"c:\Users\LucianCebuc\AERO\app\(admin)\settings.tsx",
    r"c:\Users\LucianCebuc\AERO\app\chat\[rideId].tsx",
    r"c:\Users\LucianCebuc\AERO\components\feature\DriverRequestCard.tsx",
]

# Pattern to match JSX text contents: >Text Content<
jsx_text_pattern = re.compile(r'>\s*([^<{]+?[a-zA-ZăâîșțĂÂÎȘȚ].*?)\s*<')

# Pattern to match string literals inside code: 'string' or "string"
string_literal_pattern = re.compile(r'[\'"]([^\'"]*?[ăâîșțĂÂÎȘȚ][^\'"]*?)[\'"]')

extracted_strings = {}

for filepath in FILES_TO_SCAN:
    if not os.path.exists(filepath):
        print(f"Skipping missing file: {filepath}")
        continue
        
    print(f"Scanning: {filepath}")
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    file_strings = set()
    
    # Extract JSX text
    for match in jsx_text_pattern.finditer(content):
        text = match.group(1).strip()
        if text and not text.startswith('{') and not text.endswith('}'):
            file_strings.add(text)
            
    # Extract string literals
    for match in string_literal_pattern.finditer(content):
        text = match.group(1).strip()
        if text:
            file_strings.add(text)
            
    if file_strings:
        extracted_strings[os.path.basename(filepath)] = sorted(list(file_strings))

output_path = r"c:\Users\LucianCebuc\AERO\scratch\extracted_strings.json"
os.makedirs(os.path.dirname(output_path), exist_ok=True)
with open(output_path, 'w', encoding='utf-8') as f:
    json.dump(extracted_strings, f, indent=2, ensure_ascii=False)

print(f"Extraction complete! Saved to {output_path}")
